### PR TITLE
Executing external script on cbQuestion

### PR DESCRIPTION
**Pull request by tebajanga**

### CODE REVIEW

File: modules/cbQuestion/actions/qactions.php [hunk 1]
```diff
@@ -301,5 +301,24 @@ public function delMViewWF() {
 		$rdo['notify'] = $smarty->fetch('applicationmessage.tpl');
 		echo json_encode($rdo);
 	}
+
+	public function executeScript() {
+		$record = $this->checkQIDParam();
+		$dbUser = isset($_REQUEST['db_user']) ? vtlib_purify($_REQUEST['db_user']) : '';
+		$dbPass = isset($_REQUEST['db_pass']) ? vtlib_purify($_REQUEST['db_pass']) : '';
+		$dbName = isset($_REQUEST['db_name']) ? vtlib_purify($_REQUEST['db_name']) : '';
```
Comment (joebordes) [inline modules/cbQuestion/actions/qactions.php]: these three variables should be read from config.inc.php, so change it here and eliminate them from the JS call

File: modules/cbQuestion/actions/qactions.php [hunk 2]
```diff
@@ -301,5 +301,24 @@ public function delMViewWF() {
...
+		$tablename = isset($_REQUEST['tablename']) ? vtlib_purify($_REQUEST['tablename']) : '';
+		$scriptPath = isset($_REQUEST['script_path']) ? vtlib_purify($_REQUEST['script_path']) : '';
+
+		$command = sprintf("".$scriptPath." %s %s %s %s %s", $dbUser, $dbPass, $dbName, $tablename, $record);
+		$output = shell_exec($command);
```
Comment (joebordes) [inline modules/cbQuestion/actions/qactions.php]: we have to add a list of authorized scripts here as this is to dangerous to leave open like this. I propose a simple json/xml or a database table with the list of authorized scripts

File: modules/cbQuestion/actions/qactions.php [hunk 3]
```diff
@@ -301,5 +301,31 @@ public function delMViewWF() {
 		$rdo['notify'] = $smarty->fetch('applicationmessage.tpl');
 		echo json_encode($rdo);
 	}
+
+	public function executeScript() {
+		$record = $this->checkQIDParam();
+		$db_username = $dbconfig['db_username'];
```
Comment (joebordes) [inline modules/cbQuestion/actions/qactions.php]: don't you have to add a "global $dbconfig" here for that array to be available?

Comment (tebajanga) [reply modules/cbQuestion/actions/qactions.php]: @joebordes , I included the global variables.

**Comment by joebordes** [comment]
thanks!!

**Pull request merged**